pkg/gatewayserver/redis/registry.go

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redis
+
+import (
+	"context"
+
+	ttnredis "go.thethings.network/lorawan-stack/pkg/redis"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/unique"
+)
+
+// GatewayConnectionStatsRegistry implements the GatewayConnectionStatsRegistry interface.
+type GatewayConnectionStatsRegistry struct {
+	Redis *ttnredis.Client
+}
+
+func (r *GatewayConnectionStatsRegistry) key(uid string) string {
```

Comment:
Let's turn this into `key(ids ttnpb.GatewayIdentifiers) string` and construct the uid in here.

Reply:
This would require that the context is passed as well. Also, this is also not done in other places in the code. (e.g. https://github.com/TheThingsNetwork/lorawan-stack/blob/e116caedb47be8e7a049dde5369acb790aed03ca/pkg/applicationserver/io/packages/redis/registry.go#L58-L62)